server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/process/TagAggregationOperator.java

```diff
@@ -94,13 +81,11 @@ public TagAggregationOperator(
     this.maxReturnSize = maxReturnSize;
   }
 
-  @Override
-  public OperatorContext getOperatorContext() {
-    return operatorContext;
-  }
-
   @Override
   public TsBlock next() {
+    if (!prepareInput()) {
+      return null;
+    }
```

Comment:
It seems that you still forgot to move prepareInput into the below while-loop which is processOneRow method, just like you did in AggregationOperator.